File: 01.JS-Basic/EXAMS/exams April (2) 2019/CinemaTickets.js
function cinemaTickets(input) {

    let movieName = input[0]
    let freeTickets = Number(input[1])
    let currentFilmTickets = 0
    let totalTickets = 0
    let studentTickets = 0
    let standartTickets = 0
    let kidTickets = 0
    let i = 2

    while (movieName !== "Finish") {

        while (currentFilmTickets < freeTickets) {

            let current = input[i]

            if (current === "End") {
                console.log(`${movieName} - ${((currentFilmTickets / freeTickets * 100)).toFixed(2)}% full.`)
                i++
                movieName = String(input[i])
                i++
                freeTickets = Number(input[i])
                totalTickets += currentFilmTickets
                currentFilmTickets = 0
            }
            else if (current === "student") {
                studentTickets++
                currentFilmTickets++
            } else if (current === "standard") {
                standartTickets++
                currentFilmTickets++
            } else if (current === "kid") {
                kidTickets++
                currentFilmTickets++
            }
            if (currentFilmTickets >= freeTickets) {
                console.log(`${movieName} - ${((currentFilmTickets / freeTickets * 100)).toFixed(2)}% full.`)
                i++
                movieName = String(input[i])
                i++
                freeTickets = Number(input[i])
                totalTickets += currentFilmTickets
                currentFilmTickets = 0
            }
            i++
        }
        i++
    }
    console.log(`Total tickets: ${totalTickets}`);
    console.log(`${((studentTickets / totalTickets) * 100).toFixed(2)}% student tickets.`);
    console.log(`${((standartTickets / totalTickets) * 100).toFixed(2)}% standard tickets.`);
    console.log(`${((kidTickets / totalTickets) * 100).toFixed(2)}% kids tickets.`);
}
cinemaTickets(["Taxi","10","standard","kid","student","student","standard","standard","End","Scary Movie","6","student","student","student","student","student","student","Finish"])

File: JAVA-Basic/JavaBasic/src/NestedLoops/CinemaTickets.java
package NestedLoops;

import java.util.Scanner;

public class CinemaTickets {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        String input = scan.nextLine();
        int totalTickets = 0;
        int studentTickets = 0;
        int standardTickets = 0;
        int kidTickets = 0;

        while (!input.equals("Finish")) {
            String movie = input;
            int freeSpaces = Integer.parseInt(scan.nextLine());
            int tickets = 0;
            String command = scan.nextLine();

            while (!command.equals("End")) {
                String ticketType = command;
                tickets++;

                switch (ticketType) {
                    case "student":
                        studentTickets++;
                        break;
                    case "standard":
                        standardTickets++;
                        break;
                    case "kid":
                        kidTickets++;
                        break;
                }
                if (tickets == freeSpaces) {
                    break;
                }
                command = scan.nextLine();
            }
            totalTickets += tickets;
            System.out.printf("%s - %.2f%% full.%n", movie, tickets * 1.0 / freeSpaces * 100);
            input = scan.nextLine();
        }
        System.out.println("Total tickets: " + totalTickets);
        System.out.printf("%.2f%% student tickets.%n", studentTickets * 1.0 / totalTickets * 100);
        System.out.printf("%.2f%% standard tickets.%n", standardTickets * 1.0 / totalTickets * 100);
        System.out.printf("%.2f%% kids tickets.%n", kidTickets * 1.0 / totalTickets * 100);
    }
}
